Pcap++/header/PcapDevice.h

```diff
@@ -18,6 +18,79 @@ namespace pcpp
 	// Forward Declaration - required for IPcapDevice::matchPacketWithFilter
 	class GeneralFilter;
 
+	namespace internal
+	{
+		/// @class PcapHandle
+		/// @brief A wrapper class for pcap_t* which is the libpcap packet capture descriptor.
+		/// This class is used to manage the lifecycle of the pcap_t* object
+		class PcapHandle
+		{
+		public:
+			/// @brief Creates an empty handle.
+			PcapHandle() = default;
+			/// @brief Creates a handle from the provided pcap descriptor.
+			/// @param pcapDescriptor The pcap descriptor to wrap.
+			explicit PcapHandle(pcap_t* pcapDescriptor);
+
+			PcapHandle(const PcapHandle&) = delete;
+			PcapHandle(PcapHandle&& other) noexcept;
+
+			PcapHandle& operator=(const PcapHandle&) = delete;
+			PcapHandle& operator=(PcapHandle&& other) noexcept;
+			PcapHandle& operator=(std::nullptr_t) noexcept;
+
+			~PcapHandle();
+
+			/// @brief Check if the handle is not null.
+			/// @return True if the handle is not null, false otherwise.
+			bool hasValue() const
+			{
+				return m_PcapDescriptor != nullptr;
+			}
+
+			/// @brief Access the underlying pcap descriptor.
+			/// @return The pcap descriptor.
+			pcap_t* value() const
+			{
+				return m_PcapDescriptor;
+			}
+
+			/// @brief Releases ownership of the handle and returns the pcap descriptor.
+			/// @return The pcap descriptor or nullptr if no handle is owned.
+			pcap_t* release();
+
+			/// @brief Helper function to retrieve the last error string for this handle.
+			/// @return The last error string.
+			std::string getLastError() const;
+
+			/// @brief Helper function to retrieve a view of the last error string for this handle.
+			/// @return A view of the last error string.
+			/// @remarks This function is more efficient than getLastError() as it does not copy the string.
+			char const* getLastErrorView() const;
```

Comment:
Hmm, it would probably be better to keep the `char const*` version as it avoids a copy and the user can always convert it to a string if it needs to be saved.

Reply:
Removed `getLastError` in c1d0af8. Left the `getLastErrorView` with the `View` suffix to emphasize that the returned pointer is a view and should be saved if keeping it is required.

Reply:
Why don't we want to return a string?
Errors are usually not in the critical path so coping a `char*` to `std::string` shouldn't be an issue...

Reply:
I chose to keep the view as its the more performant version as it does not have a heap allocation. Converting the view `char*` to string is trivial if we need it somewhere, and the string version could always be added back if its needed in too many places.

Reply:
So let's rename it to `getLastError()`?

Reply:
I guess we can do that. If we need to add a string return type later  it will need renaming again, but I guess its not a big issue since its an internal class.

Reply:
Changed e481d1c